src/ipa_pc/data_structures.rs

```diff
@@ -235,6 +235,41 @@ impl<G: AffineCurve> ToBytes for Proof<G> {
     }
 }
 
+/// `BatchProof` generated by batching scheme.
+/// Refers to the "Multi-poly multi-point assertions" section of the "Recursive SNARKs based on Marlin" document
+#[derive(Derivative)]
+#[derivative(
+    Default(bound = ""),
+    Hash(bound = ""),
+    Clone(bound = ""),
+    Debug(bound = "")
+)]
+pub struct BatchProof<G: AffineCurve> {
+
+    /// Muli-poly single-point proof
+    pub proof: Proof<G>,
+
+    /// Commitment of the h(X) polynomial
+    pub batch_commitment: G, 
+
+    /// Values: v_i = p_i(x), where x is fresh random challenge
+    pub batch_values: Vec<G::ScalarField>
```

Comment:
As commented below in the prover: maybe better to do a label-based mapping for the batch_values?

Reply:
Done